File: ClientUnity/Assets/Scripts/Behaviours/LocalPlayer.cs
using UnityEngine;
using System.Collections;

public class LocalPlayer : MonoBehaviour
{
    public GameObject ChargeButton;
    public GameObject ShockButton;
    public GameObject BlockButton;

    public Sprite ShockDeselected;
    public Sprite ShockSelected;
    public Sprite NukeDeselected;
    public Sprite NukeSelected;

    internal void UpdateShockButton(int power)
    {
        ToggleTuner checkBehaviour = ShockButton.GetComponent<ToggleTuner>();
        if (power >= Configs.NUKE_POWER)
        {
            checkBehaviour.DeselectedSprite = NukeDeselected;
            checkBehaviour.SelectedSprite = NukeSelected;
        }
        else
        {
            checkBehaviour.DeselectedSprite = ShockDeselected;
            checkBehaviour.SelectedSprite = ShockSelected;
        }
        if (checkBehaviour.IsSelected)
        {
            checkBehaviour.BackgroundObject.sprite = checkBehaviour.SelectedSprite;
        }
        else
        {
            checkBehaviour.BackgroundObject.sprite = checkBehaviour.DeselectedSprite;
        }
    }

    public void OnReadyClicked()
    {
        Debug.Log("clicked ready");
        JSONObject data = JSONObject.Create();
        data.AddField("name", GameManager.Instance.LocalName);
        GameManager.Instance.Emit(IOTypes.E_READY, data);
    }

    public void OnChargeClicked()
    {
        Debug.Log("clicked Charge");
        JSONObject data = JSONObject.Create();
        data.AddField("name", GameManager.Instance.LocalName);
        GameManager.Instance.Emit(IOTypes.E_CHARGE, data);
    }

    public void OnShockClicked()
    {
        Debug.Log("clicked Shock");
        Player playerBehaviour = gameObject.GetComponent<Player>();
        if (playerBehaviour.Power == 0)
        {
            Debug.Log("but not enough power, block checked instead");
            StartCoroutine(ForcePressBlock());
        }
        else if (playerBehaviour.Power >= Configs.NUKE_POWER)
        {
            JSONObject data = JSONObject.Create();
            data.AddField("name", GameManager.Instance.LocalName);
            GameManager.Instance.Emit(IOTypes.E_NUKE, data);
        }
        else
        {
            JSONObject data = JSONObject.Create();
            data.AddField("name", GameManager.Instance.LocalName);
            GameManager.Instance.Emit(IOTypes.E_SHOCK, data);
        }
    }

    IEnumerator ForcePressBlock()
    {
        yield return new WaitForSeconds(0.3f);
        BlockButton.GetComponent<ToggleTuner>().OnClicked();
    }

    public void OnBlockClicked()
    {
        Debug.Log("clicked Block");
        JSONObject data = JSONObject.Create();
        data.AddField("name", GameManager.Instance.LocalName);
        GameManager.Instance.Emit(IOTypes.E_BLOCK, data);
    }

}

File: CCclient/assets/Script/Behaviours/LocalPlayer.js
cc.Class({
    extends: cc.Component,

    properties: {
        chargeNode: {
            default: null,
            type: cc.Node
        },
        shockNode: {
            default: null,
            type: cc.Node
        },
        blockNode: {
            default: null,
            type: cc.Node
        }
    },

    updateShockButton(power) {
        let checkBehaviour = this.shockNode.getComponent('CheckBehaviour');
        if (power >= NUKE_POWER) {
            checkBehaviour.textLabel.getComponent(cc.Label).string = "HUGE STINKY\nFART";
        } else {
            checkBehaviour.textLabel.getComponent(cc.Label).string = "FART";
        }
    },

    onReadyClicked () {
        console.log(`clicked ready`);
        window.GM.emit(reqs.READY, {
            pid: window.GM.localPid,
            method: window.GM.localMethod
        });
    },

    onChargeClicked () {
        console.log(`clicked Charge`);
        window.GM.emit(reqs.CHARGE, {
            pid: window.GM.localPid,
            method: window.GM.localMethod
        });
    },

    onShockClicked () {
        console.log(`clicked Shock`);
        const playerBehaviour = this.node.getComponent('PlayerBehaviour');
        if (playerBehaviour.power == 0) {
            console.log(`but not enough power, block checked instead`);
            this.scheduleOnce(function() {
                this.blockNode.getComponent(cc.Toggle).check();
            }, 0.3);
        } else if (playerBehaviour.power >= NUKE_POWER) {
            window.GM.emit(reqs.NUKE, {
                pid: window.GM.localPid,
                method: window.GM.localMethod
            });
        } else {
            window.GM.emit(reqs.SHOCK, {
                pid: window.GM.localPid,
                method: window.GM.localMethod
            });
        }
    },

    onBlockClicked () {
        console.log(`clicked Block`);
        window.GM.emit(reqs.BLOCK, {
            pid: window.GM.localPid,
            method: window.GM.localMethod
        });
    }

});
